lib/esbonio/esbonio/lsp/sphinx/__init__.py

```diff
@@ -730,6 +733,38 @@ def get_doctree(
             # self.logger.debug(traceback.format_exc())
             return None
 
+    def get_initial_doctree(self, uri: str) -> Optional[Any]:
+        """Return the initial doctree corresponding to the specified document.
+
+        Parameters
+        ----------
+        uri:
+           Returns the doctree that corresponds with the given uri.
+        """
+
+        if self.app is None or self.app.env is None:
+            return None
+
+        fspath = Uri.to_fs_path(uri)
+        if not fspath.startswith(self.app.srcdir):
+            # Fallback to standard rst behavior.
+            return super().get_initial_doctree(uri)
+
+        self.app.env.prepare_settings(fspath)
+
+        try:
+            with disable_expansions():
+                doctree = read_doc(self.app, self.app.env, fspath)
```

Comment:
`read_doc` will be [deprecated](https://github.com/sphinx-doc/sphinx/commit/f9dce5764edd62b280904325cfecb4f962dbbd9b) as of v5.0. The suggested alternative is the `read_doc` method on the current `Builder`, but at this point it may be better just to write our own implementation.